pages/lib/extract_df.py

```diff
@@ -399,66 +399,63 @@ def create_df(lst, file_name):
-    df[name] = df[name] * 0.0929
-
-
-def speed(df, name):
-    df[name] = df[name] * 196.85039370078738
-
-
-def visibility(df, name):
-    df[name] = df[name] * 0.6215
-
-
-def enthalpy(df, name):
-    df[name] = df[name] * 0.0004
+def convert_SI_to_IP(df: pd.DataFrame, name: str) -> None:
+    """Convert SI to IP based on column name."""
+    if name not in df.columns:
+        return
+    match name:
+        case (
+            ColNames.DBT
+            | ColNames.DPT
+            | ColNames.T_WB
+            | ColNames.T_DP
+            | ColNames.UTCI_SUN_WIND
+            | ColNames.UTCI_NO_SUN_WIND
+            | ColNames.UTCI_SUN_NO_WIND
+            | ColNames.UTCI_NO_SUN_NO_WIND
+            | ColNames.ADAPTIVE_COMFORT
+            | ColNames.ADAPTIVE_CMF_80_LOW
+            | ColNames.ADAPTIVE_CMF_80_UP
+            | ColNames.ADAPTIVE_CMF_90_LOW
+            | ColNames.ADAPTIVE_CMF_90_UP
+        ):
+            df[name] = df[name] * 1.8 + 32
+
+        case ColNames.P_ATM | ColNames.P_VAP | ColNames.P_SAT:
+            df[name] = df[name] * 0.000145038
+
+        case (
+            ColNames.EXTR_HOR_RAD
+            | ColNames.HOR_IR_RAD
+            | ColNames.GLOB_HOR_RAD
+            | ColNames.DIR_NOR_RAD
+            | ColNames.DIF_HOR_RAD
+        ):
+            df[name] = df[name] * 0.3169983306
+
+        case ColNames.GLOB_HOR_ILL | ColNames.DIR_NOR_ILL | ColNames.DIF_HOR_ILL:
+            df[name] = df[name] * 0.0929
+
+        case ColNames.ZLUMI:
+            df[name] = df[name] * 0.0929
+
+        case ColNames.WIND_SPEED:
+            df[name] = df[name] * 196.85039370078738
+
+        case ColNames.VIS:
+            df[name] = df[name] * 0.6215
+
+        case ColNames.EH:
+            df[name] = df[name] * 0.000429923
 
 
 def convert_data(df, mapping_json):
-    convert_t_to_f(df, ColNames.ADAPTIVE_COMFORT)
-    convert_t_to_f(df, ColNames.ADAPTIVE_CMF_80_LOW)
-    convert_t_to_f(df, ColNames.ADAPTIVE_CMF_80_UP)
-    convert_t_to_f(df, ColNames.ADAPTIVE_CMF_90_LOW)
-    convert_t_to_f(df, ColNames.ADAPTIVE_CMF_90_UP)
-
     mapping_dict = json.loads(mapping_json)
-    for key in json.loads(mapping_json):
-        if ColNames.CONVERSION_FUNCTION in mapping_dict[key]:
-            conversion_function_name = mapping_dict[key][ColNames.CONVERSION_FUNCTION]
-            if conversion_function_name is not None:
-                conversion_function = globals()[conversion_function_name]
-                conversion_function(df, key)
-    return json.dumps(mapping_dict)
-
-
-def convert_t_to_f(df: pd.DataFrame, name: str):
-    """Convert temperature from Celsius to Fahrenheit in-place for a given column.
-
-    Args:
-        df: The DataFrame containing the temperature column.
-        name: Column name to convert.
-
-    Returns:
-        None. The DataFrame is modified in-place.
-    """
-    df[name] = df[name] * 1.8 + 32
+    for key, meta in mapping_dict.items():
+        if meta.get(ColNames.CONVERSION_FUNCTION):
```

Comment:
this is confusing, why are we still using this method of checking if the conversion function exists in the maping dictionary when we are already checking for the case in the convert_SI_to_IP? we should remove the "conversion_function" all togehter from the mapping dictionary, the, pass the key and the df to the convert_SI_to_IP function. If the key matches any of the cases then the converted value is returned otherwise the function just returns the original column untouched